src/builder/create_components.rs

```diff
@@ -112,18 +194,18 @@ pub enum CreateSelectMenuKind {
     Channel { channel_types: Option<Vec<ChannelType>> },
 }
 
+#[derive(Serialize, Deserialize)]
+struct CreateSelectMenuKindJson {
+    #[serde(rename = "type")]
+    kind: u8,
+    options: Option<Vec<CreateSelectMenuOption>>,
+    channel_types: Option<Vec<ChannelType>>,
+}
+
 impl Serialize for CreateSelectMenuKind {
     fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        #[derive(Serialize)]
-        struct Json<'a> {
-            #[serde(rename = "type")]
-            kind: u8,
-            options: Option<&'a [CreateSelectMenuOption]>,
-            channel_types: Option<&'a [ChannelType]>,
-        }
-
         #[rustfmt::skip]
```

Comment:
Changing out this section might be a bit dirty, and it might be better to just have two separate types on each side if it is better to have slices while serializing. I read [here](https://users.rust-lang.org/t/problem-with-serde-serialize-of-slice-of-structs/28070/3) that it's probably best to use a `Vec`, though I don't really understand the rabbit hold of lifetimes/references while deserializing, and this was an easy fix.